src/main/scala/is/hail/methods/Aggregators.scala

```diff
@@ -495,3 +497,34 @@ class TakeByAggregator[T](var t: Type, var f: (Any) => Any, var n: Int)(implicit
 
   def copy() = new TakeByAggregator(t, f, n)
 }
+
+class LinearRegressionAggregator(var xsF: (Any) => Any, var nxs: Int) extends TypedAggregator[Any] {
+  var combiner = new LinearRegressionCombiner(nxs)
+
+  def seqOp(x: Any) = {
+    if (x != null) {
+      val y = x.asInstanceOf[Double]
```

Comment:
can you explain whats going on here? the variables are confusing me.

Reply:
An Annotation is passed. `x` was a bad name. I changed it.